### PR TITLE
Move login out as service and add httpintercept for login prompt

### PR DESCRIPTION
**Pull request by ryanjdew**
Moved the login logic to it's own module.
Added an http intercept to bring up the login prompt when a 401 is hit.
Allows for an easy switch between full login page or a modal login prompt.
Changed nodejs proxy to require logins for REST API GETs, to match the default required login for search that previously was enforced on the front-end by default.

### CODE REVIEW

**Comment by paxtonhare** [comment]
@ryanjdew Is there a simple way to keep the logout button and user icon and still use the modal or fullscreen logins?

To test I commented out the <ml-user...> in root.html and that seemed to enable the login.service to do its magic.

But I didn't see a way to keep the logout button w/o hacking on the user directive or manually creating it myself.

I'm thinking this pull request might also need a small blurb in the README on how to swap it around.

**Comment by ryanjdew** [comment]
@paxtonhare, I added the blurb in the README for setting the appropriate variable. 

I also fixed the top right corner to have a link to login instead of the login boxes when a login mode other than 'top-right' is set.

File: app/templates/ui/app/user/user-dir.html [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
-      <input type="text" placeholder="Username" class="form-control input-sm" ng-model="ctrl.username">
-    </div>
-    <div class="form-group">
-      <input type="password" placeholder="Password" class="form-control input-sm" ng-model="ctrl.password">
-    </div>
-    <button type="submit" class="btn btn-success btn-sm">Sign in</button>
-  </form>
+  <div class="navbar-form navbar-right" ng-if="!ctrl.userService.currentUser()" >
+    <login ng-if="ctrl.loginService.loginMode() === 'top-right'" mode="inline"></login>
+    <span ng-if="ctrl.loginService.loginMode() !== 'top-right'" class="profile-link"><a ng-click="ctrl.loginService.loginPrompt()">Login</a></span>
```
Comment (paxtonhare) [inline app/templates/ui/app/user/user-dir.html]: If you change this:
<a ng-click="ctrl.loginService.loginPrompt()">Login</a>

to this:

<a href ng-click="ctrl.loginService.loginPrompt()">Login</a>

then the cursor will change to the pointer cursor vs the text cursor w/o you having to set a style.

**Comment by paxtonhare** [comment]
When _loginMode === 'full'  I'm seeing some weirdness with failed logins. Each time I do a failed login my url gets crazier:

http://localhost:3000/login?state=root.login&params=%7B%22state%22:%22root.login%22,%22params%22:%22%7B%5C%22state%5C%22:%5C%22root.login%5C%22,%5C%22params%5C%22:%5C%22%7B%5C%5C%5C%22state%5C%5C%5C%22:%5C%5C%5C%22root.search%5C%5C%5C%22,%5C%5C%5C%22params%5C%5C%5C%22:%5C%5C%5C%22%7B%7D%5C%5C%5C%22%7D%5C%22%7D%22%7D

Also, there is no indication that my login failed for any of the modes ('full', 'modal', 'top-right'). I simply just see the form again.

**Comment by paxtonhare** [comment]
The /create route isn't forcing an auth prompt. Instead it's just showing up as empty in the ui-view

**Comment by ryanjdew** [comment]
@paxtonhare , I believe I've worked through the remaining issues with the login redirects and the create route not forcing authentication.

**Comment by grtjn** [comment]
Related to #118 and #119

**Pull request merged**